askama_derive/src/generator.rs

```diff
@@ -493,47 +490,100 @@ fn compile_time_escape<'a>(expr: &Expr<'a>, escaper: &str) -> Option<Writable<'a
 }
 
 #[derive(Clone, Default)]
-struct LocalMeta {
+struct LocalVariableMeta {
     refs: Option<String>,
     initialized: bool,
 }
 
-impl LocalMeta {
-    fn initialized() -> Self {
-        Self {
+#[derive(Clone)]
+struct LocalCallerMeta<'a> {
+    def: &'a Call<'a>,
+    call_ctx: Context<'a>,
+}
+
+#[derive(Clone)]
+enum LocalMeta<'a> {
+    /// Normal variable
+    Variable(LocalVariableMeta),
+
+    /// This special variable is a caller alias. It's another name for caller().
+    CallerAlias(LocalCallerMeta<'a>),
+
+    /// Represents a "negative" local variable. Meaning: When the resolve methods
+    /// encounters a negative on its path down the stack of scopes, it will immediately
+    /// return without result. This is required to "block out" variables outside of a certain scope
+    Negative,
+}
+
+impl<'a> LocalMeta<'a> {
+    /// Variable declaration only - no value yet.
+    const fn var_decl() -> Self {
+        Self::Variable(LocalVariableMeta {
+            refs: None,
+            initialized: false,
+        })
+    }
+
+    /// Variable definition - fully initialized.
+    const fn var_def() -> Self {
+        Self::Variable(LocalVariableMeta {
             refs: None,
             initialized: true,
-        }
+        })
     }
 
-    fn with_ref(refs: String) -> Self {
-        Self {
+    /// Variable referencing another
+    const fn var_with_ref(refs: String) -> Self {
+        Self::Variable(LocalVariableMeta {
```

Comment:
Are we sure in this case the referenced variable was initialized?

Reply:
That's a question I can't answer, since this is just a small refactoring to what [is there currently](https://github.com/askama-rs/askama/blob/a3f913af7029e8da224c425d7932dbe35c4f084c/askama_derive/src/generator.rs#L509). Or did you find a mistake in my refactoring?

Reply:
No nevermind, maybe something we'll need to worry about later on. Since I was reviewing your code, I just thought about this case.